### PR TITLE
comments & eip-3541

### PR DESCRIPTION
**Pull request by krlosMata**
This PR does the following:
- implements [eip-3541](https://eips.ethereum.org/EIPS/eip-3541)
- split invalid errors into more specific
- update few comments

### CODE REVIEW

File: main/opcodes/create-terminate-context.zkasm [hunk 1]
```diff
@@ -537,12 +537,17 @@ opRETURNcreate:
 
     ; set bytecode length
     $ => A          :MLOAD(createContractAddress)
+
+    ; check bytecode first byte != 0xEF
+                    :CALL(checkBytecodeStartsEF)
```
Comment (ignasirv) [inline main/opcodes/create-terminate-context.zkasm:542]: You can make the MLOAD and the CALL at the same line:
$ => A          :MLOAD(createContractAddress), CALL(checkBytecodeStartsEF)
One step less

File: main/process-tx.zkasm [hunk 2]
```diff
@@ -446,9 +446,15 @@ endDeploy:
         ; only when first context ends on deploy
         ; If length is 0 do not modify state-tree
         C                               :JMPZ(handleGas)
+
         ; save offset memory and length to compute hash bytecode
         E                               :MSTORE(memOffsetLinearPoseidon)
         C                               :MSTORE(memSizeLinearPoseidon)
+
+        ; check bytecode first byte != 0xEF
+                                        :CALL(checkBytecodeStartsEF)
```
Comment (ignasirv) [inline main/process-tx.zkasm:455]: Can do it in on line MSTORE + CALL

**Review by laisolizq — APPROVED** [review]
:smile_cat:

**Review by ignasirv — APPROVED** [review]
💯

**Pull request merged**